### DISCUSSION TITLE
Unable to detect both gyro and temp. sensor - good electrical contacts

### DISCUSSION THREAD

**Discussion by delight-f** [mp-se/gravitymon#194] [Q&A] [answered]
Issue: 

1. Both temperature sensor and gyro unable to be detected
2. Able to flash v2.0.0. beta 3 but unable to actually function

Hardware: Opensource Distilling board v.2.6.9, ESPS2-Mini dev board [Dev Board aliexpress link](https://www.aliexpress.com/item/1005006157693055.html)

Tried so far:

1. Flashed latest build via downloaded Thorrack Brewflasher - 2.0.0 beta 3. Flash completes, but board cannot start into WiFi config mode.
2. Downgraded to 1.4.0 - able to boot into config mode, but gyro and DS18B20 not able to detected
   a. Electrical continuity between gyro I2C bus checked, same for temp sensor output and input pin.
   b. DS18B20 changed three times

On the right side of the board, working from bottom up is VBUS, GND, 16, 18. (I have used this board before for my own project, working well - it seems these pins are mapped to ESP32 GPIO, not actual pin numbers). On the left, working from bottom up is 9, 5, 3, EN.

Thoughts:

I wonder if the input pins for the temp sensor and I2C bus are not mapped correctly for this device - maybe it is a clone? If you agree, could you point me in the right direction for re-mapping these input pins?

**Comment by mp-se** [comment, marked as answer]
The pins used are the same location as on the esp8266 so if the board is created according to the standard iSpindle pinout there should not be any issues. I havent tested the opensource distilling (i use the cherry philip board).  You can find the used pins in the main.hpp file. 

If you do an firmware update the device needs a reset for the i2c to work again., dont really know why. On the temp sensor you can try the build in the dev branch to see if that works for the temp sensor. I have switched the library to a different onewire driver that seams to work better. Dont know if that has to do with clone temp sensors or something else.

**Reply by delight-f** [reply]
Thank you for your quick response. I really appreciate it! I am inexperienced in Github; do you mean to try the firmware binary in the /bin folder in the dev branch i.e. [https://github.com/mp-se/gravitymon/blob/dev/bin/firmware32s2.bin]()

If so, I gave this a go. Which is nice because I am running 2.0.0 with the new UI (very nice btw!) but still no gyro or temp sensor. Must be the poor quality clone sensor effect again!

**Reply by mp-se** [reply]
Try the iSpindle software as well, if that also fails then its certain a hardware issue.

**Comment by gizmocuz** [comment]
I am sorry to bump this old question.

I have the same issue as above with a iSpindel running on a ESP8266

I successfully compiled and uploaded the firmware (2.2.0 (beta-2). and also tried a precompiled (and older) version, but I keep getting

**"No gyro is detected. Try to reboot / power-off. If this persists, check for hardware issues."**

Of course I rebooted, powered-off etc.

Building/Flashing the latest iSpindel firmware (7.3.3) and the Gyro/Temp sensor are detected

Can you point me in a direction on how I could solve this?
